docs/src/reference/experimental/loop-contracts.md

```diff
@@ -114,33 +116,81 @@ Mathematical induction is the working principle here. (1) establishes the base c
 Therefore, the invariant must hold after the loop execution for any number of iterations. The invariant, together with the negation of the loop guard,
 must be sufficient to establish subsequent assertions. If it is not, the abstraction is too imprecise and the user must supply a stronger invariant.
 
-To illustrate the key idea, we show how Kani abstracts the loop in `simple_loop_with_loop_contracts` as a non-loop block:
+To illustrate the key idea, we show how Kani abstracts the loop in `simple_loop_with_loop_invariants` as a non-loop block:
 ``` Rust
-assert!(x >= 1) // check loop invariant for the base case.
+assert!(x >= 1) // check loop-invariant for the base case.
 x = kani::any();
 kani::assume(x >= 1);
 if x > 1 {
     // proof path 1:
-    //   both loop guard and loop invariant are satisfied.
+    //   both loop guard and loop-invariant are satisfied.
     x = x - 1;
-    assert!(x >= 1); // check that loop invariant is inductive.
+    assert!(x >= 1); // check that loop-invariant is inductive.
     kani::assume(false) // block this proof path.
 }
 // proof path 2:
-//   loop invariant is satisfied and loop guard is violated.
+//   loop-invariant is satisfied and loop guard is violated.
 assert!(x == 1);
 ```
-That is, we assume that we are in an arbitrary iteration after checking that the loop invariant holds for the base case. With the inductive hypothesis (`kani::assume(x >= 1);`),
+That is, we assume that we are in an arbitrary iteration after checking that the loop-invariant holds for the base case. With the inductive hypothesis (`kani::assume(x >= 1);`),
 we will either enter the loop (proof path 1) or leave the loop (proof path 2). We prove the two paths separately by killing path 1 with `kani::assume(false);`.
 Note that all assertions after `kani::assume(false)` will be ignored as `false => p` can be deduced as `true` for any `p`.
 
-In proof path 1, we prove properties inside the loop and at last check that the loop invariant is inductive.
+In proof path 1, we prove properties inside the loop and at last check that the loop-invariant is inductive.
 
 In proof path 2, we prove properties after leaving the loop. As we leave the loop only when the loop guard is violated, the post condition of the loop can be expressed as
-`!guard && inv`, which is `x <= 1 && x >= 1` in the example. The postcondition implies `x == 1`—the property we want to prove at the end of `simple_loop_with_loop_contracts`.
+`!guard && inv`, which is `x <= 1 && x >= 1` in the example. The postcondition implies `x == 1`—the property we want to prove at the end of `simple_loop_with_loop_invariants`.
+
+## Historic values and extra variables
```

Comment:
The documentation of historic values is unrelated to this PR, so should be moved to its own PR.